Failure to bootstrap maas - apparent distribution mismatch - conjure-up sees xenial but system is bionic

## Report

- [No ] Is this problem already documented at https://docs.ubuntu.com/conjure-up/en/troubleshoot#common-spell-problems
- [No] Is conjure-up running inside a virtual machine?
- [Yes ] Is this problem reproducible with `sudo snap refresh conjure-up --edge`?

## Please provide the output of the following commands

```
which juju
juju version
2.4.3-bionic-amd64

which conjure-up
conjure-up --version
conjure-up 2.6.1

which lxc
/snap/bin/lxc config show
config: {}
/snap/bin/lxc version
Client version: 3.5
Server version: 3.5

cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.1 LTS"
```





Please attach tarball of **~/.cache/conjure-up**:

```
tar cvzf conjure-up.tar.gz ~/.cache/conjure-up
```
[conjure-up.tar.gz](https://github.com/conjure-up/conjure-up/files/2412206/conjure-up.tar.gz)



## Sosreport

Please attach a sosreport:

```
sudo apt install sosreport
sosreport
```
[sosreport-cscott-20180924175108.tar.gz](https://github.com/conjure-up/conjure-up/files/2412244/sosreport-cscott-20180924175108.tar.gz)

## What Spell was Selected?
OpenStack with NovaKVM
## What provider (aws, maas, localhost, etc)?
maas
### MAAS Users
Which version of MAAS?
2.4.2 (7034-g2f5deb8b8-0ubuntu1)
## Commands ran
conjure-up
Please outline what commands were run to install and execute conjure-up:
sudo snap install conjure-up --classic
## Additional Information

Relevant component of conjure-up.log:

```
2018-09-24 17:24:25,495 [ERROR] conjure-up/openstack-base - common.py:60 - Error bootstrapping controller: ['Creating Juju cont
roller "conjure-up-cloud-maas-24f-02c" on cloud-maas-24f', 'Looking for packaged Juju agent version 2.4.3 for amd64', 'Launchin
g controller instance(s) on cloud-maas-24f...', 'ERROR failed to bootstrap model: cannot start bootstrap instance: unexpected: 
ServerError: 400 Bad Request ({"distro_series": ["\'xenial\' is not a valid distro_series.  It should be one of: \'\', \'ubuntu
/bionic\'."]})']
```

--- a/conjureup/juju.py
+++ b/conjureup/juju.py
@@ -198,8 +198,7 @@ async def create_model():
         await controller.disconnect()
 
 
-async def bootstrap(controller, cloud, model='conjure-up', series="xenial",
-                    credential=None):
+async def bootstrap(controller, cloud, model='conjure-up', credential=None):
     """ Performs juju bootstrap
 
     If not LXD pass along the newly defined credentials
@@ -208,7 +207,6 @@ async def bootstrap(controller, cloud, model='conjure-up', series="xenial",
     controller: name of your controller
     cloud: name of local or public cloud to deploy to
     model: name of default model to create
-    series: define the bootstrap series defaults to xenial
     credential: credentials key
     """
     if app.provider.region is not None:
@@ -243,8 +241,9 @@ async def bootstrap(controller, cloud, model='conjure-up', series="xenial",
         add_config("bootstrap-timeout", app.conjurefile['bootstrap-timeout'])
     if app.conjurefile['bootstrap-to']:
         cmd.extend(["--to", app.conjurefile['bootstrap-to']])
+    if app.conjurefile['bootstrap-series']:
+        cmd.extend(["--bootstrap-series", app.conjurefile['bootstrap-series']])
 
-    cmd.extend(["--bootstrap-series", series])
     if credential is not None:
         cmd.extend(["--credential", credential])
 

--- a/conjureup/models/conjurefile.py
+++ b/conjureup/models/conjurefile.py
@@ -98,6 +98,9 @@ class Conjurefile(MeldDict):
     # Comma separate list of ips to not filter through proxy
     # no-proxy: 8.8.8.8,172.16.0.1
 
+    # Override Juju default controller series
+    # bootstrap-series: bionic
+
     # (Optional) Bundle Add
     # Add a Juju bundle fragment overlay
     # bundle-add: /home/user/my-bundle-fragment.yaml

